packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -66,6 +67,97 @@ interface RulesParameters {
   votingPeriod: number;
 }
 
+// Type guard for MainTransaction.
+const isMainTransaction = (transaction: MainTransaction): transaction is MainTransaction => {
+  try {
+    BigNumber.from(transaction.value);
+  } catch {
+    return false;
+  }
+  return (
+    typeof transaction === "object" &&
+    typeof transaction.to === "string" &&
+    ethersUtils.isAddress(transaction.to) &&
+    ethersUtils.isHexString(transaction.data) &&
+    typeof transaction.value === "string" &&
+    BigNumber.from(transaction.value).gte(0) &&
+    (transaction.operation === "0" || transaction.operation === "1")
+  );
+};
+
+// Type guard for SafeSnapSafe.
```

Comment:
I know the checks inside these functions feel a bit messy, the way you've structured these type guards is really nice. I also really appreciate how you nested the calls :). Really cool

This will also make it really easy to update if the structure of these objects changes.